interface/usergroup/mfa_u2f.php

```diff
@@ -61,94 +65,151 @@ function docancel() {
-    } catch (u2flib_server\Error $e) {
-        die(xlt('Registration error') . ': ' . text($e->getMessage()));
-    }
-    echo "<script>\n";
-    $row = sqlQuery(
-        "SELECT COUNT(*) AS count FROM login_mfa_registrations WHERE " .
-        "`user_id` = ? AND `name` = ?",
-        array($userid, $_POST['form_name'])
-    );
-    if (empty($row['count'])) {
-        sqlStatement(
-            "INSERT INTO login_mfa_registrations " .
-            "(`user_id`, `method`, `name`, `var1`, `var2`) VALUES " .
-            "(?, 'U2F', ?, ?, ?)",
-            array($userid, $_POST['form_name'], json_encode($data), '')
-        );
-    } else {
-        echo " alert(" . xlj('This key name is already in use by you. Try again.') . ");\n";
-    }
-    echo " window.location.href = 'mfa_registrations.php';\n";
-    echo "</script>\n";
-}
+        if ($action == 'reg1') {
+            list ($request, $signs) = $u2f->getRegisterData();
+            ?>
+        <!--<div class="row">
+            <div class="col-sm-12">
+              <p>
+                <?php echo xlt('This will register a new U2F USB key.'); ?>
+                <?php echo xlt('Type a name for your key, insert it into a USB port and click the Register button below.'); ?>
+                <?php echo xlt('Then press the flashing button on your key within 1 minute to complete registration.'); ?>
+              </p>
+              <table><tr><td>
+                <?php echo xlt('Please give this key a name'); ?>:
+              <input type='text' name='form_name' value='' size='16' />&nbsp;</td>
+              <td><input type='button' value='<?php echo xla('Register'); ?>' onclick='doregister()' />
+              <input type='button' value='<?php echo xla('Cancel'); ?>' onclick='docancel()'   />
+              <input type='hidden' name='form_request' value='<?php echo attr(json_encode($request)); ?>' />
+              <input type='hidden' name='form_signs'   value='<?php echo attr(json_encode($signs)); ?>' />
+              <input type='hidden' name='form_registration' value='' />
+              </td></tr></table>
+              &nbsp;<br />
+              <p>
+                <?php echo xlt('A secure (HTTPS) web connection is required for U2F. Firefox and Chrome are known to work.'); ?>
+              </p>
+              <p>
+                <?php echo xlt('For U2F support on Linux see'); ?>:
+              <a href='https://www.key-id.com/enable-fido-u2f-linux/' rel="noopener" target='_blank'>
+              https://www.key-id.com/enable-fido-u2f-linux/</a>
+              </p>
+              <p>
+                <?php echo xlt('For Firefox see'); ?>:
+              <a href='https://www.trishtech.com/2018/07/enable-fido-u2f-security-key-yubikey-in-mozilla-firefox/' rel="noopener" target='_blank'>
+              https://www.trishtech.com/2018/07/enable-fido-u2f-security-key-yubikey-in-mozilla-firefox/</a>
+              </p>
+            </div>
+        </div>-->
```

Comment:
better to remove code than comment it out (also easier to review since then can see what code you have moved)

Reply:
sorry missed it during cleanup. now removed